### PR TITLE
add PCA, pending tests

### PR DESCRIPTION
**Pull request by CJ-Wright**

### CODE REVIEW

File: xpdtools/tools.py [hunk 1]
```diff
@@ -580,3 +580,11 @@ def inner(x, *args, **kwargs):
         return func(*args, **kwargs)
 
     return inner
+
+
+def decomp(data, variance=.9):
```
Comment (sbillinge) [inline xpdtools/tools.py]: This is just returning eigenvalues, which I guess go in as QOIs, which is good.  If the user wants to plot the eigenvectors, do we want to give them that opportunity?

Comment (sbillinge) [inline xpdtools/tools.py]: brief docstring here too?

Comment (sbillinge) [inline xpdtools/tools.py]: this is only one way to decompose a matrix.....should we call this ``pca_decomp``?  Or probably more correct, ``pc_decomp``?

Comment (CJ-Wright) [reply xpdtools/tools.py]: I'm planning on setting up a kwarg to allow users to pass in a decomposition engine in so any system which works the same can be used. I did this for now for simplicity and to keep as close to example code as possible

Comment (CJ-Wright) [reply xpdtools/tools.py]: Sure, I'm not certain how to get that out of the model though.

**Comment by CJ-Wright** [comment]
Plotting the eigenvectors will need a new plotting system since we want to clear the plot each new data set, but it should be possible.

**Comment by CJ-Wright** [comment]
I think the main issue for this will be on the xpdAn side where we need a descriptor router to handle this (since we don't know how many things we want to PCA until we know what the descriptor looks like).

File: xpdtools/tools.py [hunk 2]
```diff
@@ -580,3 +581,29 @@ def inner(x, *args, **kwargs):
...
+
+    Returns
+    -------
+    eigen : ndarray
+        The eigenvalues
+    """
+    m = model(n_components, **kwargs)
+    m.fit(data)
+    eigen = m.components_
+    return eigen
```
Comment (chiahaoliu) [inline xpdtools/tools.py]:
```suggestion
    return eigen, m.transform(data)
```

**Pull request merged**